File: TruckManagementFrontEnd/src/app/types/truck.ts

export class Truck {    
    Id?: number;    
    TruckChassis!: string;   
    TruckColor!: string; 
    TruckManufacturingDate?: Date;    
    TruckManufacturingPlant!: string;
    TruckModel!: string;  
    TruckManufacturingDateF: string;    
   
   

    constructor(id:number, truckChassis:string, truckColor:string, truckManufacturingDate: Date, TruckManufacturingPlant: string, truckModel: string, truckManufacturingDateF: string)   
    {
        this.Id = id;
        this.TruckChassis = truckChassis;
        this.TruckColor = truckColor;
        this.TruckManufacturingDate = truckManufacturingDate;
        this.TruckManufacturingPlant = TruckManufacturingPlant;
        this.TruckModel = truckModel;
        this.TruckManufacturingDateF = truckManufacturingDateF; 
    }
}

File: TruckManagementWebApi/Domain/Entities/Truck.cs
using Domain;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Numerics;
using System.Runtime.ConstrainedExecution;
using System.Text;
using System.Threading.Tasks;

namespace TruckManagement.Domain.Entities
{
    public class Truck
    {
        [Key]
        public int Id { get; set; }
        public string TruckModel { get; set; }
        public DateTime TruckManufacturingDate { get; set; }
        public string TruckChassis { get; set; }
        public string TruckColor { get; set; }
        public string TruckManufacturingPlant { get; set; }

    }   
}
